### ISSUE TITLE
stdbuf -oL -eL causes "Command clang++ returned non-zero exit status -4" and other crashes

### ISSUE THREAD

**Issue by cpitclaudel** [emscripten-core/emscripten#6458]
Hi,

I'm trying to compile a simple program and I'm running into the following issue:

```bash
printf '#include<stdio.h>\nint main() { return 0; }\n' > minimal.c
emcc -v minimal.c
```

```
  INFO:root:(Emscripten: Running sanity checks)
  WARNING:root:java does not seem to exist, required for closure compiler, which is optional (define JAVA in /home/vagrant/.emscripten if you want it)
  WARNING:root:closure compiler will not be available
  clang version 5.0.0  (emscripten 1.37.37 : 1.37.37)
  Target: asmjs-unknown-emscripten
  Thread model: posix
  InstalledDir: /vagrant/z3-wasm/emsdk-portable/clang/e1.37.37_64bit
   "/vagrant/z3-wasm/emsdk-portable/clang/e1.37.37_64bit/clang" -cc1 -triple asmjs-unknown-emscripten -emit-llvm-bc -emit-llvm-uselists -disable-free -disable-llvm-verifier -discard-value-names -main-file-name minimal.c -mrelocation-model static -mthread-model posix -mdisable-fp-elim -no-integrated-as -mconstructor-aliases -v -dwarf-column-info -debugger-tuning=gdb -coverage-notes-file /tmp/tmp6BgTtU/minimal_0.gcno -nostdsysteminc -nobuiltininc -resource-dir /vagrant/z3-wasm/emsdk-portable/clang/lib/clang/5.0.0 -D __EMSCRIPTEN_major__=1 -D __EMSCRIPTEN_minor__=37 -D __EMSCRIPTEN_tiny__=37 -D _LIBCPP_ABI_VERSION=2 -Werror=implicit-function-declaration -fno-dwarf-directory-asm -fdebug-compilation-dir /tmp/emscripten -ferror-limit 19 -fmessage-length 0 -fobjc-runtime=gnustep -fdiagnostics-show-option -nobuiltininc -nostdsysteminc -isystem/vagrant/z3-wasm/emsdk-portable/emscripten/1.37.37/system/include/libcxx -isystem/vagrant/z3-wasm/emsdk-portable/emscripten/1.37.37/system/lib/libcxxabi/include -isystem/vagrant/z3-wasm/emsdk-portable/emscripten/1.37.37/system/include/compat -isystem/vagrant/z3-wasm/emsdk-portable/emscripten/1.37.37/system/include -isystem/vagrant/z3-wasm/emsdk-portable/emscripten/1.37.37/system/include/SSE -isystem/vagrant/z3-wasm/emsdk-portable/emscripten/1.37.37/system/include/libc -isystem/vagrant/z3-wasm/emsdk-portable/emscripten/1.37.37/system/lib/libc/musl/arch/emscripten -isystem/vagrant/z3-wasm/emsdk-portable/emscripten/1.37.37/system/local/include -disable-O0-optnone -isystem/vagrant/z3-wasm/emsdk-portable/emscripten/1.37.37/system/include/SDL -o /tmp/tmp6BgTtU/minimal_0.o -x c minimal.c
  clang -cc1 version 5.0.0 based upon LLVM 5.0.0 default target x86_64-unknown-linux-gnu
  #include "..." search starts here:
  #include <...> search starts here:
   /vagrant/z3-wasm/emsdk-portable/emscripten/1.37.37/system/include/libcxx
   /vagrant/z3-wasm/emsdk-portable/emscripten/1.37.37/system/lib/libcxxabi/include
   /vagrant/z3-wasm/emsdk-portable/emscripten/1.37.37/system/include/compat
   /vagrant/z3-wasm/emsdk-portable/emscripten/1.37.37/system/include
   /vagrant/z3-wasm/emsdk-portable/emscripten/1.37.37/system/include/SSE
   /vagrant/z3-wasm/emsdk-portable/emscripten/1.37.37/system/include/libc
   /vagrant/z3-wasm/emsdk-portable/emscripten/1.37.37/system/lib/libc/musl/arch/emscripten
   /vagrant/z3-wasm/emsdk-portable/emscripten/1.37.37/system/local/include
   /vagrant/z3-wasm/emsdk-portable/emscripten/1.37.37/system/include/SDL
  End of search list.
  INFO:root:generating system library: dlmalloc.bc... (this will be cached in "/home/vagrant/.emscripten_cache/asmjs/dlmalloc.bc" for subsequent builds)
  INFO:root: - ok
  INFO:root:generating system library: libc.bc... (this will be cached in "/home/vagrant/.emscripten_cache/asmjs/libc.bc" for subsequent builds)
  ERROR:root:compiler frontend failed to generate LLVM bitcode, halting
  Traceback (most recent call last):
    File "/vagrant/z3-wasm/emsdk-portable/emscripten/1.37.37/emcc", line 10, in <module>
      python_selector.run(__file__, profile=True)
    File "/vagrant/z3-wasm/emsdk-portable/emscripten/1.37.37/tools/python_selector.py", line 38, in run
      sys.exit(run_by_import(filename, main) if on_allowed_version() else run_by_subprocess(filename))
    File "/vagrant/z3-wasm/emsdk-portable/emscripten/1.37.37/tools/python_selector.py", line 13, in run_by_import
      return getattr(importlib.import_module(os.path.basename(filename)), main)()
    File "/vagrant/z3-wasm/emsdk-portable/emscripten/1.37.37/emcc.py", line 1512, in run
      extra_files_to_link += system_libs.calculate([f for _, f in sorted(temp_files)] + extra_files_to_link, in_temp, stdout_=None, stderr_=None, forced=forced_stdlibs)
    File "/vagrant/z3-wasm/emsdk-portable/emscripten/1.37.37/tools/system_libs.py", line 549, in calculate
      libfile = shared.Cache.get(name, do_create, extension=suffix)
    File "/vagrant/z3-wasm/emsdk-portable/emscripten/1.37.37/tools/cache.py", line 104, in get
      temp = creator()
    File "/vagrant/z3-wasm/emsdk-portable/emscripten/1.37.37/tools/system_libs.py", line 547, in do_create
      ret = create(name)
    File "/vagrant/z3-wasm/emsdk-portable/emscripten/1.37.37/tools/system_libs.py", line 145, in create_libc
      return build_libc(libname, libc_files, args)
    File "/vagrant/z3-wasm/emsdk-portable/emscripten/1.37.37/tools/system_libs.py", line 87, in build_libc
      run_commands(commands)
    File "/vagrant/z3-wasm/emsdk-portable/emscripten/1.37.37/tools/system_libs.py", line 30, in run_commands
      pool.map_async(call_process, commands, chunksize=1).get(999999)
    File "/usr/lib/python2.7/multiprocessing/pool.py", line 567, in get
      raise self._value
  Exception: Command '/usr/bin/python /vagrant/z3-wasm/emsdk-portable/emscripten/1.37.37/emcc.py /vagrant/z3-wasm/emsdk-portable/emscripten/1.37.37/system/lib/libc/musl/src/stdio/__towrite.c -o /tmp/tmp6BgTtU/__towrite.c.o -I /vagrant/z3-wasm/emsdk-portable/emscripten/1.37.37/system/lib/libc/musl/src/internal -I /vagrant/z3-wasm/emsdk-portable/emscripten/1.37.37/system/lib/libc/musl/arch/js -Werror -Wno-return-type -Wno-parentheses -Wno-ignored-attributes -Wno-shift-count-overflow -Wno-shift-negative-value -Wno-dangling-else -Wno-unknown-pragmas -Wno-shift-op-parentheses -Wno-string-plus-int -Wno-logical-op-parentheses -Wno-bitwise-op-parentheses -Wno-visibility -Wno-pointer-sign -Wno-absolute-value -Wno-empty-body -Os -fno-builtin' returned non-zero exit status 1
  Traceback (most recent call last):
    File "/vagrant/z3-wasm/emsdk-portable/emscripten/1.37.37/emcc.py", line 2812, in <module>
      sys.exit(run())
    File "/vagrant/z3-wasm/emsdk-portable/emscripten/1.37.37/emcc.py", line 1469, in run
      safe_move(temp_file, bitcode_target)
    File "/vagrant/z3-wasm/emsdk-portable/emscripten/1.37.37/tools/shared.py", line 2712, in safe_move
      shutil.move(src, dst)
    File "/usr/lib/python2.7/shutil.py", line 302, in move
      copy2(src, real_dst)
    File "/usr/lib/python2.7/shutil.py", line 130, in copy2
      copyfile(src, dst)
    File "/usr/lib/python2.7/shutil.py", line 83, in copyfile
      with open(dst, 'wb') as fdst:
  IOError: [Errno 2] No such file or directory: '/tmp/tmp6BgTtU/pclose.c.o'
  /vagrant/z3-wasm/emsdk-portable/clang/e1.37.37_64bit/opt: /tmp/tmp6BgTtU/open_wmemstream.c.o: error: Could not open input file: No such file or directory
  ERROR:root:Failed to run llvm optimizations: 
  WARNING:root:Note: Input file "/tmp/tmp6BgTtU/open_wmemstream.c.o" did not exist.
  error: unable to rename temporary '/tmp/tmp6BgTtU/putc.c.o-63963d4e' to output file '/tmp/tmp6BgTtU/putc.c.o': 'No such file or directory'
  1 error generated.
  ERROR:root:compiler frontend failed to generate LLVM bitcode, halting
  error: unable to open output file '/tmp/tmp6BgTtU/__uflow.c.o': 'No such file or directory'
  1 error generated.
  ERROR:root:compiler frontend failed to generate LLVM bitcode, halting
  error: unable to open output file '/tmp/tmp6BgTtU/vdprintf.c.o': 'No such file or directory'
  1 error generated.
  ERROR:root:compiler frontend failed to generate LLVM bitcode, halting
  error: unable to open output file '/tmp/tmp6BgTtU/vscanf.c.o': 'No such file or directory'
  1 error generated.
  ERROR:root:compiler frontend failed to generate LLVM bitcode, halting
```

I'm not sure how to proceed. Is there a way to get progress output during the libc.bc generation?  Is there a way to limit the number of processes used during that phase?

Thanks!

**Comment by cpitclaudel** [comment]
Here's another backtrace:

```
INFO:root:(Emscripten: Running sanity checks)
WARNING:root:java does not seem to exist, required for closure compiler, which is optional (define JAVA in /home/vagrant/.emscripten if you want it)
WARNING:root:closure compiler will not be available
clang version 5.0.0  (emscripten 1.37.37 : 1.37.37)
Target: asmjs-unknown-emscripten
Thread model: posix
InstalledDir: /vagrant/z3-wasm/emsdk-portable/clang/e1.37.37_64bit
 "/vagrant/z3-wasm/emsdk-portable/clang/e1.37.37_64bit/clang" -cc1 -triple asmjs-unknown-emscripten -emit-llvm-bc -emit-llvm-uselists -disable-free -disable-llvm-verifier -discard-value-names -main-file-name minimal.c -mrelocation-model static -mthread-model posix -mdisable-fp-elim -no-integrated-as -mconstructor-aliases -v -dwarf-column-info -debugger-tuning=gdb -coverage-notes-file /tmp/tmpcFvkMc/minimal_0.gcno -nostdsysteminc -nobuiltininc -resource-dir /vagrant/z3-wasm/emsdk-portable/clang/lib/clang/5.0.0 -D __EMSCRIPTEN_major__=1 -D __EMSCRIPTEN_minor__=37 -D __EMSCRIPTEN_tiny__=37 -D _LIBCPP_ABI_VERSION=2 -Werror=implicit-function-declaration -fno-dwarf-directory-asm -fdebug-compilation-dir /tmp/emscripten -ferror-limit 19 -fmessage-length 214 -fobjc-runtime=gnustep -fdiagnostics-show-option -nobuiltininc -nostdsysteminc -isystem/vagrant/z3-wasm/emsdk-portable/emscripten/1.37.37/system/include/libcxx -isystem/vagrant/z3-wasm/emsdk-portable/emscripten/1.37.37/system/lib/libcxxabi/include -isystem/vagrant/z3-wasm/emsdk-portable/emscripten/1.37.37/system/include/compat -isystem/vagrant/z3-wasm/emsdk-portable/emscripten/1.37.37/system/include -isystem/vagrant/z3-wasm/emsdk-portable/emscripten/1.37.37/system/include/SSE -isystem/vagrant/z3-wasm/emsdk-portable/emscripten/1.37.37/system/include/libc -isystem/vagrant/z3-wasm/emsdk-portable/emscripten/1.37.37/system/lib/libc/musl/arch/emscripten -isystem/vagrant/z3-wasm/emsdk-portable/emscripten/1.37.37/system/local/include -disable-O0-optnone -isystem/vagrant/z3-wasm/emsdk-portable/emscripten/1.37.37/system/include/SDL -o /tmp/tmpcFvkMc/minimal_0.o -x c minimal.c
clang -cc1 version 5.0.0 based upon LLVM 5.0.0 default target x86_64-unknown-linux-gnu
#include "..." search starts here:
#include <...> search starts here:
 /vagrant/z3-wasm/emsdk-portable/emscripten/1.37.37/system/include/libcxx
 /vagrant/z3-wasm/emsdk-portable/emscripten/1.37.37/system/lib/libcxxabi/include
 /vagrant/z3-wasm/emsdk-portable/emscripten/1.37.37/system/include/compat
 /vagrant/z3-wasm/emsdk-portable/emscripten/1.37.37/system/include
 /vagrant/z3-wasm/emsdk-portable/emscripten/1.37.37/system/include/SSE
 /vagrant/z3-wasm/emsdk-portable/emscripten/1.37.37/system/include/libc
 /vagrant/z3-wasm/emsdk-portable/emscripten/1.37.37/system/lib/libc/musl/arch/emscripten
 /vagrant/z3-wasm/emsdk-portable/emscripten/1.37.37/system/local/include
 /vagrant/z3-wasm/emsdk-portable/emscripten/1.37.37/system/include/SDL
End of search list.
INFO:root:generating system library: libc.bc... (this will be cached in "/home/vagrant/.emscripten_cache/asmjs/libc.bc" for subsequent builds)
#0 0x000000000150e69a llvm::sys::PrintStackTrace(llvm::raw_ostream&) (/vagrant/z3-wasm/emsdk-portable/clang/e1.37.37_64bit/clang+0x150e69a)
#1 0x000000000150c75e llvm::sys::RunSignalHandlers() (/vagrant/z3-wasm/emsdk-portable/clang/e1.37.37_64bit/clang+0x150c75e)
#2 0x000000000150c8ac SignalHandler(int) (/vagrant/z3-wasm/emsdk-portable/clang/e1.37.37_64bit/clang+0x150c8ac)
#3 0x00002b7f1b50e390 __restore_rt (/lib/x86_64-linux-gnu/libpthread.so.0+0x11390)
#4 0x000000000245a245 clang::Sema::ActOnVariableDeclarator(clang::Scope*, clang::Declarator&, clang::DeclContext*, clang::TypeSourceInfo*, clang::LookupResult&, llvm::MutableArrayRef<clang::TemplateParameterList*>, bool&, llvm::ArrayRef<clang::BindingDecl*>) (/vagrant/z3-wasm/emsdk-portable/clang/e1.37.37_64bit/clang+0x245a245)
#5 0x00000000024673b8 clang::Sema::HandleDeclarator(clang::Scope*, clang::Declarator&, llvm::MutableArrayRef<clang::TemplateParameterList*>) (/vagrant/z3-wasm/emsdk-portable/clang/e1.37.37_64bit/clang+0x24673b8)
#6 0x0000000002467a59 clang::Sema::ActOnDeclarator(clang::Scope*, clang::Declarator&) (/vagrant/z3-wasm/emsdk-portable/clang/e1.37.37_64bit/clang+0x2467a59)
#7 0x0000000002192490 clang::Parser::ParseDeclarationAfterDeclaratorAndAttributes(clang::Declarator&, clang::Parser::ParsedTemplateInfo const&, clang::Parser::ForRangeInit*) (/vagrant/z3-wasm/emsdk-portable/clang/e1.37.37_64bit/clang+0x2192490)
#8 0x00000000021a1f54 clang::Parser::ParseDeclGroup(clang::ParsingDeclSpec&, unsigned int, clang::SourceLocation*, clang::Parser::ForRangeInit*) (/vagrant/z3-wasm/emsdk-portable/clang/e1.37.37_64bit/clang+0x21a1f54)
#9 0x000000000217ca9a clang::Parser::ParseDeclOrFunctionDefInternal(clang::Parser::ParsedAttributesWithRange&, clang::ParsingDeclSpec&, clang::AccessSpecifier) (/vagrant/z3-wasm/emsdk-portable/clang/e1.37.37_64bit/clang+0x217ca9a)
#10 0x000000000217d1c1 clang::Parser::ParseDeclarationOrFunctionDefinition(clang::Parser::ParsedAttributesWithRange&, clang::ParsingDeclSpec*, clang::AccessSpecifier) [clone .part.150] [clone .constprop.182] (/vagrant/z3-wasm/emsdk-portable/clang/e1.37.37_64bit/clang+0x217d1c1)
#11 0x000000000217d1ff clang::Parser::ParseDeclarationOrFunctionDefinition(clang::Parser::ParsedAttributesWithRange&, clang::ParsingDeclSpec*, clang::AccessSpecifier) (/vagrant/z3-wasm/emsdk-portable/clang/e1.37.37_64bit/clang+0x217d1ff)
#12 0x00000000021841c8 clang::Parser::ParseExternalDeclaration(clang::Parser::ParsedAttributesWithRange&, clang::ParsingDeclSpec*) (/vagrant/z3-wasm/emsdk-portable/clang/e1.37.37_64bit/clang+0x21841c8)
#13 0x00000000021850b5 clang::Parser::ParseTopLevelDecl(clang::OpaquePtr<clang::DeclGroupRef>&) (/vagrant/z3-wasm/emsdk-portable/clang/e1.37.37_64bit/clang+0x21850b5)
#14 0x0000000002178acb clang::ParseAST(clang::Sema&, bool, bool) (/vagrant/z3-wasm/emsdk-portable/clang/e1.37.37_64bit/clang+0x2178acb)
#15 0x0000000001d8bc57 clang::CodeGenAction::ExecuteAction() (/vagrant/z3-wasm/emsdk-portable/clang/e1.37.37_64bit/clang+0x1d8bc57)
#16 0x0000000001a2e84e clang::FrontendAction::Execute() (/vagrant/z3-wasm/emsdk-portable/clang/e1.37.37_64bit/clang+0x1a2e84e)
#17 0x00000000019fcfe6 clang::CompilerInstance::ExecuteAction(clang::FrontendAction&) (/vagrant/z3-wasm/emsdk-portable/clang/e1.37.37_64bit/clang+0x19fcfe6)
#18 0x0000000001aba692 clang::ExecuteCompilerInvocation(clang::CompilerInstance*) (/vagrant/z3-wasm/emsdk-portable/clang/e1.37.37_64bit/clang+0x1aba692)
#19 0x000000000097c1e8 cc1_main(llvm::ArrayRef<char const*>, char const*, void*) (/vagrant/z3-wasm/emsdk-portable/clang/e1.37.37_64bit/clang+0x97c1e8)
#20 0x000000000093125b main (/vagrant/z3-wasm/emsdk-portable/clang/e1.37.37_64bit/clang+0x93125b)
#21 0x00002b7f1c1df830 __libc_start_main (/lib/x86_64-linux-gnu/libc.so.6+0x20830)
#22 0x0000000000979e59 _start (/vagrant/z3-wasm/emsdk-portable/clang/e1.37.37_64bit/clang+0x979e59)
Stack dump:
0.	Program arguments: /vagrant/z3-wasm/emsdk-portable/clang/e1.37.37_64bit/clang -cc1 -triple asmjs-unknown-emscripten -emit-llvm-bc -emit-llvm-uselists -disable-free -disable-llvm-verifier -discard-value-names -main-file-name getwchar.c -mrelocation-model static -mthread-model posix -mdisable-fp-elim -no-integrated-as -mconstructor-aliases -dwarf-column-info -debugger-tuning=gdb -coverage-notes-file /tmp/tmpcFvkMc/getwchar.c.gcno -nostdsysteminc -nobuiltininc -resource-dir /vagrant/z3-wasm/emsdk-portable/clang/lib/clang/5.0.0 -D __EMSCRIPTEN_major__=1 -D __EMSCRIPTEN_minor__=37 -D __EMSCRIPTEN_tiny__=37 -D _LIBCPP_ABI_VERSION=2 -I /vagrant/z3-wasm/emsdk-portable/emscripten/1.37.37/system/lib/libc/musl/src/internal -I /vagrant/z3-wasm/emsdk-portable/emscripten/1.37.37/system/lib/libc/musl/arch/js -Os -Werror=implicit-function-declaration -Werror -Wno-return-type -Wno-parentheses -Wno-ignored-attributes -Wno-shift-count-overflow -Wno-shift-negative-value -Wno-dangling-else -Wno-unknown-pragmas -Wno-shift-op-parentheses -Wno-string-plus-int -Wno-logical-op-parentheses -Wno-bitwise-op-parentheses -Wno-visibility -Wno-pointer-sign -Wno-absolute-value -Wno-empty-body -fno-dwarf-directory-asm -fdebug-compilation-dir /vagrant/z3-wasm/emsdk-portable/emscripten/1.37.37 -ferror-limit 19 -fmessage-length 214 -fno-builtin -fno-inline-functions -fobjc-runtime=gnustep -fdiagnostics-show-option -vectorize-loops -vectorize-slp -nobuiltininc -nostdsysteminc -isystem/vagrant/z3-wasm/emsdk-portable/emscripten/1.37.37/system/include/libcxx -isystem/vagrant/z3-wasm/emsdk-portable/emscripten/1.37.37/system/lib/libcxxabi/include -isystem/vagrant/z3-wasm/emsdk-portable/emscripten/1.37.37/system/include/compat -isystem/vagrant/z3-wasm/emsdk-portable/emscripten/1.37.37/system/include -isystem/vagrant/z3-wasm/emsdk-portable/emscripten/1.37.37/system/include/SSE -isystem/vagrant/z3-wasm/emsdk-portable/emscripten/1.37.37/system/include/libc -isystem/vagrant/z3-wasm/emsdk-portable/emscripten/1.37.37/system/lib/libc/musl/arch/emscripten -isystem/vagrant/z3-wasm/emsdk-portable/emscripten/1.37.37/system/local/include -disable-O0-optnone -isystem/vagrant/z3-wasm/emsdk-portable/emscripten/1.37.37/system/include/SDL -disable-llvm-optzns -o /tmp/tmpcFvkMc/getwchar.c.o -x c /vagrant/z3-wasm/emsdk-portable/emscripten/1.37.37/system/lib/libc/musl/src/stdio/getwchar.c 
1.	/vagrant/z3-wasm/emsdk-portable/emscripten/1.37.37/system/include/libc/stdio.h:55:25: current parser token ';'
clang: error: unable to execute command: Segmentation fault (core dumped)
clang: error: clang frontend command failed due to signal (use -v to see invocation)
clang version 5.0.0  (emscripten 1.37.37 : 1.37.37)
Target: asmjs-unknown-emscripten
Thread model: posix
InstalledDir: /vagrant/z3-wasm/emsdk-portable/clang/e1.37.37_64bit
clang: note: diagnostic msg: PLEASE submit a bug report to http://llvm.org/bugs/ and include the crash backtrace, preprocessed source, and associated run script.
clang: note: diagnostic msg: 
********************

PLEASE ATTACH THE FOLLOWING FILES TO THE BUG REPORT:
Preprocessed source(s) and associated run script(s) are located at:
clang: note: diagnostic msg: /tmp/getwchar-ff7fa2.c
clang: note: diagnostic msg: /tmp/getwchar-ff7fa2.sh
clang: note: diagnostic msg: 

********************
ERROR:root:compiler frontend failed to generate LLVM bitcode, halting
Traceback (most recent call last):
  File "/vagrant/z3-wasm/emsdk-portable/emscripten/1.37.37/emcc", line 10, in <module>
    python_selector.run(__file__, profile=True)
  File "/vagrant/z3-wasm/emsdk-portable/emscripten/1.37.37/tools/python_selector.py", line 38, in run
    sys.exit(run_by_import(filename, main) if on_allowed_version() else run_by_subprocess(filename))
  File "/vagrant/z3-wasm/emsdk-portable/emscripten/1.37.37/tools/python_selector.py", line 13, in run_by_import
    return getattr(importlib.import_module(os.path.basename(filename)), main)()
  File "/vagrant/z3-wasm/emsdk-portable/emscripten/1.37.37/emcc.py", line 1512, in run
    extra_files_to_link += system_libs.calculate([f for _, f in sorted(temp_files)] + extra_files_to_link, in_temp, stdout_=None, stderr_=None, forced=forced_stdlibs)
  File "/vagrant/z3-wasm/emsdk-portable/emscripten/1.37.37/tools/system_libs.py", line 549, in calculate
    libfile = shared.Cache.get(name, do_create, extension=suffix)
  File "/vagrant/z3-wasm/emsdk-portable/emscripten/1.37.37/tools/cache.py", line 104, in get
    temp = creator()
  File "/vagrant/z3-wasm/emsdk-portable/emscripten/1.37.37/tools/system_libs.py", line 547, in do_create
    ret = create(name)
  File "/vagrant/z3-wasm/emsdk-portable/emscripten/1.37.37/tools/system_libs.py", line 145, in create_libc
    return build_libc(libname, libc_files, args)
  File "/vagrant/z3-wasm/emsdk-portable/emscripten/1.37.37/tools/system_libs.py", line 87, in build_libc
    run_commands(commands)
  File "/vagrant/z3-wasm/emsdk-portable/emscripten/1.37.37/tools/system_libs.py", line 30, in run_commands
    pool.map_async(call_process, commands, chunksize=1).get(999999)
  File "/usr/lib/python2.7/multiprocessing/pool.py", line 567, in get
    raise self._value
Exception: Command '/usr/bin/python /vagrant/z3-wasm/emsdk-portable/emscripten/1.37.37/emcc.py /vagrant/z3-wasm/emsdk-portable/emscripten/1.37.37/system/lib/libc/musl/src/stdio/getwchar.c -o /tmp/tmpcFvkMc/getwchar.c.o -I /vagrant/z3-wasm/emsdk-portable/emscripten/1.37.37/system/lib/libc/musl/src/internal -I /vagrant/z3-wasm/emsdk-portable/emscripten/1.37.37/system/lib/libc/musl/arch/js -Werror -Wno-return-type -Wno-parentheses -Wno-ignored-attributes -Wno-shift-count-overflow -Wno-shift-negative-value -Wno-dangling-else -Wno-unknown-pragmas -Wno-shift-op-parentheses -Wno-string-plus-int -Wno-logical-op-parentheses -Wno-bitwise-op-parentheses -Wno-visibility -Wno-pointer-sign -Wno-absolute-value -Wno-empty-body -Os -fno-builtin' returned non-zero exit status 1
Traceback (most recent call last):
  File "/vagrant/z3-wasm/emsdk-portable/emscripten/1.37.37/emcc.py", line 2812, in <module>
vagrant@ubuntu-xenial:/tmp/emscripten$     sys.exit(run())
  File "/vagrant/z3-wasm/emsdk-portable/emscripten/1.37.37/emcc.py", line 1469, in run
    safe_move(temp_file, bitcode_target)
  File "/vagrant/z3-wasm/emsdk-portable/emscripten/1.37.37/tools/shared.py", line 2712, in safe_move
    shutil.move(src, dst)
  File "/usr/lib/python2.7/shutil.py", line 302, in move
    copy2(src, real_dst)
  File "/usr/lib/python2.7/shutil.py", line 130, in copy2
    copyfile(src, dst)
  File "/usr/lib/python2.7/shutil.py", line 83, in copyfile
    with open(dst, 'wb') as fdst:
IOError: [Errno 2] No such file or directory: '/tmp/tmpcFvkMc/setlinebuf.c.o'
error: unable to open output file '/tmp/tmpcFvkMc/getwc.c.o': 'No such file or directory'
1 error generated.
ERROR:root:compiler frontend failed to generate LLVM bitcode, halting
error: unable to open output file '/tmp/tmpcFvkMc/funlockfile.c.o': 'No such file or directory'
1 error generated.
ERROR:root:compiler frontend failed to generate LLVM bitcode, halting
error: unable to open output file '/tmp/tmpcFvkMc/fputws.c.o': 'No such file or directory'
1 error generated.
ERROR:root:compiler frontend failed to generate LLVM bitcode, halting
error: unable to open output file '/tmp/tmpcFvkMc/fgetws.c.o': 'No such file or directory'
1 error generated.
ERROR:root:compiler frontend failed to generate LLVM bitcode, halting
error: unable to open output file '/tmp/tmpcFvkMc/freopen.c.o': 'No such file or directory'
1 error generated.
ERROR:root:compiler frontend failed to generate LLVM bitcode, halting
```

**Comment by cpitclaudel** [comment]
I have traced the error, and the original report was incomplete. Sorry.

The crash came from running prefixed with `stdbuf -oL -eL`.

I used that command to display emcc's progress while piping its output into a log (using `tee`).

**Comment by kripken** [comment]
> The crash came from running prefixed with `stdbuf -oL -eL`.

I don't follow, sorry - what does "prefixed" mean here, and what is `stdbuf`?

> a way to limit the number of processes used during that phase

Yes, you can set `EMCC_CORES=1` in the env for example.

To debug this further if necessary, you can set `EMCC_DEBUG=1` in the env, which should print out the compilation commands as it runs them.

**Comment by cpitclaudel** [comment]
> I don't follow, sorry - what does "prefixed" mean here, and what is stdbuf?

Yes, sorry, this wasn't a great bug report :/ I was running into all sorts of issues while running in a VM.
Here are a bit more details:

I was hoping to make simple, reproducible WASM builds of Z3 (https://github.com/cpitclaudel/z3.wasm).  
I set up a VM using Vagrant, and wrote a script that printed high-level progress info, plus output from each command.
I wanted to save the output of emcc, emconfigure, etc to a log file in addition to displaying it, so I wrote something like `emcc … |& tee -a logfile`.  The buffering that this introduced, though, made the output unpleasant (large batches of output were printed at once).
From https://unix.stackexchange.com/a/25378/55157 I learnt that one way to disable buffering was the `stdbuf` program, so I rewrote the command above to `stdbuf -oL -eL emcc … |& tee -a logfile`.
This worked (output was smooth), but I started seeing the error listed above.
